it/sender_util.go

```diff
@@ -2,246 +2,106 @@ package it
+func processFile(file string) ([]url.Values, error) {
+	content, err := os.ReadFile(file)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read file[%v]: %w", file, err)
 	}
 
-	countryCode := "DE"
-
-	fmt.Println(strings.NewReader(params.Encode()))
-
-	serverHost := os.Getenv("BRUM_SERVER_HOST")
-	serverPort := os.Getenv("BRUM_SERVER_PORT")
-	req, _ := http.NewRequest("POST", fmt.Sprintf("http://%v:%v/beacon/catcher", serverHost, serverPort), strings.NewReader(params.Encode()))
-	req.Header.Add("User-Agent", uaStr)
-	req.Header.Add("CF-IPCountry", countryCode)
-	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
-
-	resp, err := client.Do(req)
-
+	var rows []map[string]any
+	err = json.Unmarshal(content, &rows)
 	if err != nil {
-		fmt.Println("Client err")
-		fmt.Printf("%s", err)
+		return nil, fmt.Errorf("unable to parse json[%s]: %w", content, err)
 	}
+	var result []url.Values
+	for i, row := range rows {
+		beaconData, ok := row["beacon_data"].(string)
+		if !ok {
+			return nil, fmt.Errorf("unable to parse beacon data[%s], file: %s, line: %d, %w", row["beacon_data"], file, i+1, err)
+		}
+		var beaconDataMap map[string]string
+		err = json.Unmarshal([]byte(beaconData), &beaconDataMap)
 
-	defer resp.Body.Close()
-
-	body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			log.Printf("Bad beacon_data in file: %v", err)
+			continue
+		}
 
-	if err != nil {
-		fmt.Println("Body read err")
-		fmt.Printf("%s", err)
+		result = append(result, makeUrlValues(beaconDataMap))
 	}
-
-	fmt.Println(string(body))
+	return result, nil
 }
 
-func httpPostFormNewStyle(params url.Values, client *http.Client, cnt int) {
-	// fmt.Println(params.Get("request_headers"))
-
-	var headers map[string][]string
-
-	unmrErr := json.Unmarshal([]byte(params.Get("request_headers")), &headers)
-
-	if unmrErr != nil {
-		fmt.Println("Bad headers JSON")
-		fmt.Println(unmrErr)
+func scanFile(fileName string) ([]url.Values, error) {
+	file, err := os.Open(fileName)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read file[%v]: %w", fileName, err)
 	}
 
-	countryCode := headers["Cf-Ipcountry"][0]
-	cityName := headers["Cf-Ipcity"][0]
-
-	fmt.Println(strings.NewReader(params.Encode()))
-
-	serverHost := os.Getenv("BRUM_SERVER_HOST")
-	serverPort := os.Getenv("BRUM_SERVER_PORT")
-	req, _ := http.NewRequest("POST", fmt.Sprintf("http://%v:%v/beacon/catcher", serverHost, serverPort), strings.NewReader(params.Encode()))
+	defer file.Close()
+	var row map[string]string
+	scanner := bufio.NewScanner(file)
 
-	if uaStr, ok := headers["User-Agent"]; ok {
-		req.Header.Add("User-Agent", uaStr[0])
-	} else {
-		req.Header.Add("User-Agent", "")
+	var result []url.Values
+	for scanner.Scan() {
+		err = json.Unmarshal([]byte(scanner.Text()), &row)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse json[%s]: %w", []byte(scanner.Text()), err)
+		}
+		beaconData := row
```

Comment:
do not copy variable row to beaconData. Rename variable row to beaconData. Move 
```
var beaconData map[string]string
```
inside the for.